packages/Manager/src/main/kotlin/nz/govt/eop/consumers/WaterAllocationConsumer.kt

```diff
@@ -0,0 +1,92 @@
+package nz.govt.eop.consumers
+
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
+import java.util.function.BiFunction
+import mu.KotlinLogging
+import mu.withLoggingContext
+import nz.govt.eop.messages.WaterAllocationMessage
+import nz.govt.eop.si.jooq.tables.WaterAllocations.Companion.WATER_ALLOCATIONS
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.common.TopicPartition
+import org.jooq.*
+import org.jooq.impl.DSL.*
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.kafka.annotation.KafkaListener
+import org.springframework.kafka.core.KafkaTemplate
+import org.springframework.kafka.listener.DeadLetterPublishingRecoverer
+import org.springframework.kafka.listener.DefaultErrorHandler
+import org.springframework.stereotype.Component
+import org.springframework.util.backoff.ExponentialBackOff
+
+const val WATER_ALLOCATION_TOPIC_NAME = "water.allocation"
+
+@Component
+class WaterAllocationConsumer(
+    @Autowired val context: DSLContext,
+    private val kafkaTemplate: KafkaTemplate<String, WaterAllocationMessage>
+) {
+
+  private val logger = KotlinLogging.logger {}
+
+  @KafkaListener(
+      topics = [WATER_ALLOCATION_TOPIC_NAME], id = "nz.govt.eop.consumers.water-allocation")
+  fun processMessage(allocation: WaterAllocationMessage) {
+
+    val recievedAtUTC = allocation.receivedAt.atOffset(ZoneOffset.UTC)
```

Comment:
This could be neater handled by a Jooq converter but I've left for a future tidy up.